lib/internal/Magento/Framework/DataObject.php

```diff
@@ -387,22 +387,26 @@ public function toString($format = '')
      */
     public function __call($method, $args)
     {
-        switch (substr((string)$method, 0, 3)) {
-            case 'get':
-                $key = $this->_underscore(substr($method, 3));
-                $index = isset($args[0]) ? $args[0] : null;
-                return $this->getData($key, $index);
-            case 'set':
-                $key = $this->_underscore(substr($method, 3));
-                $value = isset($args[0]) ? $args[0] : null;
-                return $this->setData($key, $value);
-            case 'uns':
-                $key = $this->_underscore(substr($method, 3));
-                return $this->unsetData($key);
-            case 'has':
-                $key = $this->_underscore(substr($method, 3));
-                return isset($this->_data[$key]);
+        switch($method[0]) {
+            case 'g':
```

Comment:
I don't think we can check the only first letter. 
Examples: `$dataObject->greatMethod('test')` earlier were throwing exception, but now will go into this case.

Reply:
I tested and tried varied things, and reverted it back for now... performance gains are a bit less, but still very good:
Homepage without block_cache is now: 158.512655
PLP without block_cache is now: 172.213267